opencog/cython/PythonEval.cc

```diff
@@ -44,60 +52,239 @@ using namespace opencog;
 //#define DPRINTF printf
 #define DPRINTF(...)
 
+
+static const char* DEFAULT_PYTHON_MODULE_PATHS[] =
+{
+    PROJECT_BINARY_DIR"/opencog/cython", // bindings
+    PROJECT_SOURCE_DIR"/opencog/python", // opencog modules written in python
+    PROJECT_SOURCE_DIR"/tests/cython",   // for testing
+    DATADIR"/python",                    // install directory
+    #ifndef WIN32
+    "/usr/local/share/opencog/python",
+    "/usr/share/opencog/python",
+    #endif // !WIN32
+    NULL
+};
+
+
 PythonEval* PythonEval::singletonInstance = NULL;
+AtomSpace* PythonEval::singletonAtomSpace = NULL;
```

Comment:
again, in the long-run, the atomspace can't be a singleton, although, short-term, that's OK-ish.  Other atomspaces are expected to hold temporaries of various sorts, or to hold special-purpose subsets of the data to avoid clutter.  Its still rather experimental, though.